lib/kv-extreme-scale.js

```diff
@@ -275,61 +268,46 @@ function(modelName, filter, options) {
 
   function takeNextFromCache(callback) {
     const value = cache.shift();
-
-    ModelKeyComposer.parse(value, (err, parsed) => {
-      if (err && err.code == 'NO_MODEL_PREFIX') {
-        err = null;
-        parsed = {modelName: null, key: value};
-      }
-
-      if (err) return cb(err);
-
-      if (parsed.modelName !== modelName) {
-        g.warn(
-          'Warning: key scan returned a key beloging to a wrong model.' +
-            '\nExpected model name: %j' +
-            '\nActual model name:   %j' +
-            '\nThe key: %j',
-          modelName, parsed.modelName, value);
-      }
-
-      callback(null, parsed.key);
-    });
+    callback(null, value);
   }
 
   function reportEnd(callback) {
     setImmediate(callback);
   }
 
   function fetchFirst500Keys(callback) {
-    ModelKeyComposer.compose(modelName, KEY_QUERY, (err, pattern) => {
+    self._query(modelName, KEY_QUERY, (err, body) => {
       if (err) return callback(err);
-      self._query(pattern, (err, body) => {
-        if (err) return callback(err);
-        cache = body.items.map(it => it.key);
-        getNextKey(callback);
-      });
+      cache = body.items.map(it => it.key);
+      getNextKey(callback);
     });
   }
 };
 
-ExtremeScaleKVConnector.prototype._query = function(query, callback) {
+ExtremeScaleKVConnector.prototype._query =
+function(modelName, query, callback) {
+  const mapname = this.mapname(modelName);
+  const baseUrl = this._queryUrl + '/' + mapname;
   const requestOptions = {
-    baseUrl: this._queryUrl,
+    baseUrl: baseUrl,
     uri: '/',
     method: 'GET',
     qs: {query: query},
     json: true,
   };
 
-  this.request(requestOptions, (err, res, body) => {
+  this.request(modelName, requestOptions, (err, res, body) => {
     callback(err, body);
   });
 };
 
-ExtremeScaleKVConnector.prototype.request = function(options, callback) {
+ExtremeScaleKVConnector.prototype.request =
+function(modelName, options, callback) {
+  const mapname = this.mapname(modelName);
+  const baseUrl = this._url + '/' + mapname;
```

Comment:
`(options.baseUrl || this._url)`?

Then we will also need to fix `extend` so that `options.baseUrl` does not override this computed `baseUrl`.